hat/app/org/hatdex/hat/api/service/richData/RichDataService.scala

```diff
@@ -257,6 +260,27 @@ class RichDataService @Inject() (implicit ec: DalExecutionContext) {
     }
   }
 
+  def getRecord(userId: UUID,
+                recordId: UUID)(implicit db: Database): Future[EndpointData] = {
+
+    (db.run{
+      DataJson
+        .filter(r => r.recordId === recordId && r.owner === userId)
+        .result
+      } recover {
+      case e: NoSuchElementException =>
```

Comment:
this will never happen as far as I know

Reply:
ok good to know i copied it from existing code but will remove